### PR TITLE
[MPS] Support dilated Deeplab

### PR DESCRIPTION
**Pull request by fujunwei**
fix https://github.com/intel/webml-polyfill/issues/360

### CODE REVIEW

**Comment by fujunwei** [comment]
@huningxin PTAL, thanks.

File: services/ml/compilation_impl_mac_mps.mm [hunk 1]
```diff
@@ -136,21 +141,14 @@ bool CompileConv2DOrDepthwiseConv2D(
...
-    DLOG(INFO) << "  padding_code: " << padding_code;
-  } else {
-    DLOG(INFO) << "  padding_left: " << padding_left;
-    DLOG(INFO) << "  padding_right: " << padding_right;
-    DLOG(INFO) << "  padding_top: " << padding_top;
-    DLOG(INFO) << "  padding_bottom: " << padding_bottom;
+  uint32_t dilation_x = 1, dilation_y = 1;
+  if (operation.type == mojom::ATROUS_DEPTHWISE_CONV_2D ||
+      operation.type == mojom::ATROUS_CONV_2D) {
+    dilation_x = stride_width;
```
Comment (huningxin) [inline services/ml/compilation_impl_mac_mps.mm]: I would suggest to introduce `dilation_x` and `dilation_y` into `ParameterExtracterForConv`. Assigning `stride_width` to `dilation_x` looks confused.

**Comment by huningxin** [comment]
And please rename the commit message to something like "[MPS] add atrous convolution" as it is not only for DeepLab model.

**Comment by huningxin** [comment]
Thanks @fujunwei ! I finished my first round of review.

File: services/ml/compilation_impl_mac_mps.mm (hunk 1 above)
Comment (fujunwei) [reply services/ml/compilation_impl_mac_mps.mm]: Maybe it's better to reduce parameters with defining the local variable closing to usage.

Comment (huningxin) [reply services/ml/compilation_impl_mac_mps.mm]: > how about host them in a struct?

That's good point. Actually, I introduce the `ConvParams` in MKLDNN PR: https://github.com/otcshare/chromium-src/blob/135ca7e3da4be53b546ca1035fa802372a660adc/services/ml/compilation_impl.h#L34

Let me know what do you think about this approach.

Comment (fujunwei) [reply services/ml/compilation_impl_mac_mps.mm]: Got it,  i will use it after refactoring MPS implementation align new design.

**Comment by fujunwei** [comment]
PTAL again, thanks.

**Comment by huningxin** [comment]
Looks good! Thanks!

**Pull request merged**